IsraelHiking.Web/src/application/services/route-statistics.service.spec.ts

```diff
@@ -168,8 +168,8 @@ describe("RouteStatisticsService", () => {
 
         let statistics = service.getStatistics(routeData, null, null, null, false);
 
-        expect(statistics.gain).toBe(0);
-        expect(statistics.loss).toBe(0);
+        expect(statistics.gain).toBeCloseTo(3);
+        expect(statistics.loss).toBeCloseTo(-3);
```

Comment:
This is interesting to look at - the previous algorithm had smoothen a route that goes up 3 meters and then down 3 meters. It might be interesting to try and fix the route in the test to receive the same results - i.e. a route that is almost planar with a "glitch" in gps position should report a 0 gain and loss...?

Reply:
the assumption is that outliers are rare and very different from the median of the data around them - a route with only 3 points with a small glitch won't be considered an outlier under this assumption. but I'll add a test that does demonstrate outliers.

Reply:
I see, in that case a different test would be the correct solution here.